crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -781,9 +813,90 @@ impl LogsController {
             metadata_store_client,
             metadata_writer,
             async_operations: JoinSet::default(),
+            tail_updates_tx,
+            in_flight_refresh_tail: None,
         })
     }
 
+    pub fn watch_logs_tail_updates(&self) -> watch::Receiver<Arc<LogsTailUpdates>> {
+        self.tail_updates_tx.subscribe()
+    }
+
+    pub fn schedule_refresh_tails(&mut self) -> Result<(), ShutdownError> {
+        if let Some(ref task) = self.in_flight_refresh_tail {
+            if !task.is_finished() {
+                // another task is in flight
+                return Ok(());
+            }
+        }
+
+        // none
+        let logs = Arc::clone(&self.inner.current_logs);
+        let bifrost = self.bifrost.clone();
+        let metadata_store_client = self.metadata_store_client.clone();
+        let metadata_writer = self.metadata_writer.clone();
+        let tx = self.tail_updates_tx.clone();
+        let find_tail = async move {
```

Comment:
[⬇️ Low] With my operator hat on, it might make sense to wrap this in a span, and emit a metric on how long this operation takes? We haven't done a whole lot of instrumentation on these new code paths and I can imagine that being very useful in time.

Reply:
definitely

Reply:
Instrumentation is added here https://github.com/restatedev/restate/pull/2106/commits/8e0bf6bf42cf8be1c3e8b8f03ca300be5031cb0e#diff-5145e40acf9b30f1bef7be5855117b68387f1805c73fed11c9014d261ae8e943R952